packages/@webex/plugin-cc/src/services/HttpRequest.ts

```diff
@@ -0,0 +1,147 @@
+import {SUBSCRIBE_API, WCC_API_GATEWAY, WEBSOCKET_EVENT_TIMEOUT} from './constants';
+import {WebexSDK, HTTP_METHODS, SubscribeRequest, WelcomeEvent, IHttpResponse} from '../types';
+import IWebSocket from './WebSocket/types';
+import WebSocket from './WebSocket';
+import {CC_EVENTS, SubscribeResponse} from './types';
+import {EVENT} from '../constants';
+
+export type EventHandler = {(data: any): void};
+
+class HttpRequest {
+  private webSocket: IWebSocket;
+  private webex: WebexSDK;
+  private eventHandlers: Map<string, EventHandler>;
+
+  constructor(options: {webex: WebexSDK}) {
+    const {webex} = options;
+    this.webex = webex;
+    this.webSocket = new WebSocket({
+      parent: this.webex,
+    });
+    this.eventHandlers = new Map();
+
+    // Centralized WebSocket event listener
+    this.webSocket.on(EVENT, (eventData) => {
+      this.webex.logger.log(`Received event: ${eventData.type}`);
+      const handler = this.eventHandlers.get(eventData.type);
+      if (handler) {
+        handler(eventData.data);
+      }
+    });
+  }
+
+  /* This calls subscribeNotifications and establishes a websocket connection
+   * It sends the request and then listens for the Welcome event
+   * If the Welcome event is received, it resolves the promise
+   * If the Welcome event is not received, it rejects the promise
+   */
+  public async subscribeNotifications(options: {body: SubscribeRequest}): Promise<WelcomeEvent> {
+    try {
+      const {body} = options;
+      const eventType = CC_EVENTS.WELCOME;
+      const subscribeResponse: SubscribeResponse = await this.webex.request({
+        service: WCC_API_GATEWAY,
+        resource: SUBSCRIBE_API,
+        method: HTTP_METHODS.POST,
+        body,
+      });
+
+      return new Promise((resolve, reject) => {
+        const timeoutId = setTimeout(() => {
+          this.webex.logger.error('Timeout waiting for event');
+          this.eventHandlers.delete(eventType);
+          reject(new Error('Timeout waiting for event'));
+        }, WEBSOCKET_EVENT_TIMEOUT);
+
+        // Store the event handler
+        this.eventHandlers.set(eventType, (data: any) => {
```

Comment:
We can create a generic type with mandatory fields and use that for event data being received